src/commands/up.rs

```diff
@@ -458,6 +461,17 @@ impl AppSource {
     }
 }
 
+impl Display for AppSource {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            AppSource::None => write!(f, "<no source>"),
+            AppSource::File(path) => write!(f, "local app {path:?}"),
+            AppSource::OciRegistry(reference) => write!(f, "remote app {reference:?}"),
+            AppSource::Unresolvable(s) => write!(f, "unknown app source {s:?}"),
```

Comment:
Unless I've missed where you changed it, the `Unresolvable` case carries an error message, not the original `app_source` flag text from the CLI.

Reply:
Added a colon to clarify that it is a "reason". I think in practice this case is actually unreachable anyway.